packages/interactivity/src/utils.js

```diff
@@ -53,6 +58,51 @@ export function useSignalEffect( callback ) {
 	}, [] );
 }
 
+/**
+ * Returns the passed function wrapped with the current scope so it is
+ * accessible whenever the function runs. This is primarily to make the scope
+ * available inside hook callbacks.
+ *
+ * @param {Function} func The passed function.
+ * @return {Function} The wrapped function.
+ */
+const withScope = ( func ) => {
```

Comment:
I love this 🙂

---

Let's think if we should make it a public API because it could be a way to solve the problem with global event listeners:

```js
store("myPlugin", {
  callbacks: {
    init() {
      window.attachEventListener(
        "scroll",
        withScope(() => {
          const ctx = getContext(); // works!
        })
      );
    },
  },
});
```

I'll try to open that discussion today.

Reply:
There you go:

- https://github.com/WordPress/gutenberg/discussions/57815